src/backend/joanie/edx_imports/api/__init__.py

```diff
@@ -0,0 +1,45 @@
+"""Viewset for course_run in the edx_imports app."""
+
+from http import HTTPStatus
+
+from rest_framework.decorators import (
+    api_view,
+    authentication_classes,
+    permission_classes,
+)
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from joanie.core.models import CourseRun
+from joanie.core.permissions import HasAPIKey
+from joanie.core.serializers import AdminCourseRunSerializer
+
+
+@api_view(["GET"])
+@permission_classes([HasAPIKey])
+@authentication_classes([])
+def course_run_view(request: Request):
+    """
+    Look for first course_runs related to a resource_link.
+    """
+    resource_link = request.query_params.get("resource_link")
+    if resource_link is None:
+        return Response(
+            {"detail": "Query parameter `resource_link` is required."},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    course_run = (
+        CourseRun.objects.filter(resource_link__icontains=resource_link)
+        .select_related("course")
+        .first()
+    )
+
+    if not course_run:
+        return Response(
+            {"detail": "Course run not found."},
+            status=HTTPStatus.NOT_FOUND,
+        )
```

Comment:
Why not use `from rest_framework.generics.get_object_or_404` ?

Reply:
Oh you are right, I look for the constraint in the Meta class but it is on the field itself !

Reply:
But in that why I can create 2 course runs with the same course_link in my tests...

Reply:
I think we have a problem with this constraint : 

```
    def test_course_run_resource_link_unicity(self):
        resource_link = "https://moodle.test/session1"

        course_run1 = factories.CourseRunFactory(resource_link=resource_link)

        self.assertEqual(course_run1.resource_link, resource_link)

        course_run2 = factories.CourseRunFactory(resource_link=resource_link)

        self.assertEqual(course_run2.resource_link, resource_link)
```

I will open an issue with this test in order to investigate.

Reply:
As seen together, this is not a bug. The factory is creating only one `CourseRun` because the `resource_link` already exists in database.